File: Programmers/92335.py
def changeBase(n, k):
    temp = ""
    while n:
        temp += str(n % k)
        n //= k
    return temp[::-1]


def isPrime(n):
    if n < 2:
        return False
    if n == 2:
        return True
    for i in range(2, int(n ** 0.5) + 1):
        if n % i == 0:
            return False
    return True


def solution(n, k):
    if k != 10:
        n = changeBase(n, k)
    else:
        n = str(n) 
    
    arr = n.split('0')
    answer = 0
    for num in arr:
        if num != '' and isPrime(int(num)):
            answer += 1

    return answer


print(solution(3, 4))

File: Programmers/92335.js
function solution(n, k) {
  let convertedNum;

  if (k === 10) convertedNum = String(n);
  else {
    let temp = [];
    while (n >= k) {
      temp.push(n % k);
      n = Math.floor(n / k);
    }
    if (n !== 0) temp.push(n);
    temp.reverse();
    convertedNum = temp.join("");
  }

  let divided = convertedNum.split("0");
  let answer = 0;

  divided.forEach((d) => {
    if (d && isPrime(d)) answer += 1;
  });

  return answer;
}

function isPrime(n) {
  n = Number(n);
  if (n === 1) return false;
  if (n === 2) return true;

  for (let i = 2; i <= Math.floor(Math.sqrt(n)); i++) {
    if (n % i === 0) return false;
  }
  return true;
}
